rectools/models/nn/transformer_base.py

```diff
@@ -682,19 +722,22 @@ def _init_data_preparator(self) -> None:
         raise NotImplementedError()
 
     def _init_trainer(self) -> None:
-        self._trainer = Trainer(
-            max_epochs=self.epochs,
-            min_epochs=self.epochs,
-            deterministic=self.deterministic,
-            enable_progress_bar=self.verbose > 0,
-            enable_model_summary=self.verbose > 0,
-            logger=self.verbose > 0,
-            enable_checkpointing=False,
-            devices=1,
-        )
+        if self.get_trainer_func is None:
+            self._trainer = Trainer(
+                max_epochs=self.epochs,
+                min_epochs=self.epochs,
+                deterministic=self.deterministic,
+                enable_progress_bar=self.verbose > 0,
+                enable_model_summary=self.verbose > 0,
+                logger=self.verbose > 0,
+                enable_checkpointing=self.verbose > 99,
```

Comment:
Wow, why 99? Have you at least mentioned this anywhere?

Reply:
I don't mind changing the actual number. 
Functionality is very useful - it is a single checkpoint after each epoch being replaced with the last model state.
And no need for custom trainer.
Lightning actually has it enabled by default.

I wrote it in models docstring: ~Enables automatic lightning checkpointing when set to 100 or higher. This will save the most the most recent model to a single checkpoint after each epoch.`

Reply:
Probably it worths to make an additional param for it? Verbose is mostly for logging. And sometimes you don't need logs but want to have checkpoints (I think it makes sense that they are enabled by default in lightning, though it's annoying sometimes)

Reply:
removed